### PR TITLE
Make usage of std{in,out,err} compile with -preview=nosharedaccess

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: druntime/src/core/internal/parseoptions.d [hunk 1]
```diff
@@ -146,9 +146,11 @@ private:
 
 bool optError(const scope char[] msg, const scope char[] name, const(char)[] errName)
 {
+    import core.atomic : atomicLoad;
+
     version (CoreUnittest) if (inUnittest) return false;
 
-    fprintf(stderr, "%.*s %.*s option '%.*s'.\n",
+    fprintf(atomicLoad(stderr), "%.*s %.*s option '%.*s'.\n",
```
Comment (Geod24) [inline druntime/src/core/internal/parseoptions.d:153]: > src\core\internal\parseoptions.d(153): Error: none of the overloads of template `core.atomic.atomicLoad` are callable using argument types `!()(shared(_iobuf)*)`

**Comment by RazvanN7** [comment]
We have fixed our testing pipeline, so if you rebase to latest master spurious failures should be filtered out.

**Comment by RazvanN7** [comment]
It seems that the windows failures are legit:

```
src\core\internal\parseoptions.d(153): Error: none of the overloads of template `core.atomic.atomicLoad` are callable using argument types `!()(shared(_iobuf)*)`
src\core\atomic.d(88):        Candidates are: `atomicLoad(MemoryOrder ms = MemoryOrder.seq, T)(ref return scope const T val)`
src\core\atomic.d(102):                        `atomicLoad(MemoryOrder ms = MemoryOrder.seq, T)(ref return scope const shared T val)`
src\core\atomic.d(112):                        `atomicLoad(MemoryOrder ms = MemoryOrder.seq, T)(const shared ref T val)`
```

**Comment by atilaneves** [comment]
Yes, the failures are real, I just haven't worked on this since, but I will this week.

**Comment by atilaneves** [comment]
I eventually figured out that the issue is that for the Microsoft C runtime, `stderr` et al are actually functions, and since all `atomicLoad` overloads takes their parameter by `ref`, `atomicLoad(stderr)` won't work unless we also use `-preview=rvaluerefparam` which is well beyond the scope of this PR and work.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**